docs/src/main/mdoc/datatypes/statet.md

```diff
@@ -0,0 +1,244 @@
+---
+layout: docs
+title:  "StateT"
+section: "data"
+source: "core/src/main/scala/cats/data/package.scala"
+scaladoc: "#cats.data.StateT"
+---
+# StateT
+
+`StateT[F[_], S, A]` is a data type used to define state machine
+effectful computations that can be easily composed into a larger one.
```

Comment:
Can we directly reference the `State` docs here? e.g.
> `StateT` is a monad transformer for `State` that enables you to combine the state machine effects of `State` with the effects of another monad `F[_]`

In fact, it's notable that `State[A] = StateT[Eval, A]` 😁 

http://typelevel.org/cats/datatypes/state.html

Reply:
In addition to the comment (https://github.com/typelevel/cats/pull/4126/files#r810610617), if you like I can add to this first phrase a reference to `State` like:

```suggestion
`StateT[F[_], S, A]` is a data type that extends the `State` one further: it is used to define state machine computations as well, but it also allows to perform effects at each step.
`StateT` is also a `Monad`, therefore we can easily compose its values into a larger one.
```

Reply:
I see, let's iterate once more then :) 
```suggestion
StateT[F[_], S, A]` is a data type that extends the `State` one further: it is used to define state machine computations as well, but it also allows to perform computations returning a value enclosed in `F` at each step. Such capability is often called `lifting`.
In addition, `StateT` values share the same composition properties as `F` values: if `F` is a `Functor` then `StateT[F,A,B]` will also be a `Functor`. However, beware that different `StateT` operations might required different typeclass instances, refer to the [API](https://typelevel.org/cats/api/cats/data/package$$StateT$.html) for the details.
```

Reply:
hmm 🤔 how about

> `StateT[F[_], S, A]` is a data type that generalizes `State` with the ability to compose with effects in `F[_]`. Because `StateT` is defined in terms of `F`, it is a monad only if `F` is a monad. Additionally, `StateT` may acquire new capabilities via `F`: for example, if `F` is capable of error handling via `MonadThrow[F]`, then Cats derives an instance of `MonadThrow[StateT[F, S, *]]`.